### ISSUE TITLE
Form builder

### ISSUE THREAD

**Issue by BigBadOwl** [json-schema-form/json-schema-form#3]
Has anyone thought about a form builder like this one. https://github.com/formio/ngFormBuilder?

Possibly even a fork to work with json-schema-form

**Comment by nicklasb** [comment]
You mean like the JSON Schema Builder?  :-)
https://github.com/json-schema-form/json-schema-builder

**Comment by BigBadOwl** [comment]
I have seen JSON Schema Builder, but its very "raw". 
ngFormBuilder is very advanced with the ability to order fields, field sets with nesting etc. It outputs JSON and uses Form.io as it's renderer.
I think their is an opportunity to change its output format use ASF as it's renderer instead. This would be a big win for the ASF community.

**Comment by nicklasb** [comment]
Yes, but it is basically just started. The difficult part is what is done, to order stuff is not hard. 
But sure, that would be great.

However, there is another, and very important use case for use case for the JSF builder, and that is that it is thought a directive that is possible to use it as a part of an application interface, making it possible for users to dynamically change structures in the application.

For example, to enable users to add or remove fields in an application. Or add validation, for example.
